src/electron/npm/libreoffice.d.ts

```diff
@@ -0,0 +1,310 @@
+     * @param {{[name:string]:any}} arguments - the arguments for the event
+     * @returns {void}
+     */
+    sendContentControlEvent(arguments: {[name: string]: any}): void;
+
+    /**
+     * @description gets the page rect objects for the document
+     * @returns {Rect[]} - the page rect objects
+     */
+    pageRects(): Rect[];
+
+    /**
+     * @description gets the document size
+     * @returns {Size} - the size object of the document
+     */
+    size(): Size;
+
+    /**
+     * @description if the document is ready
+     * @returns {boolean}
+     */
+    isReady(): boolean;
+  }
+
+  export interface OfficeClient {
+    /**
+     * @description add an event listener
+     * @param {string} eventName - the name of the event
+     * @param {() => void} callback - the callback function
+     * @returns {void}
+     */
+    on(eventName: string, callback: () => void): void;
+
+    /**
+     * @description turn off an event listener
+     * @param {string} eventName - the name of the event
+     * @param {() => void} callback - the callback function
+     * @returns {void}
+     */
+    off(eventName: string, callback: () => void): void;
+
+    /**
+     * @description emit an event for an event listener
+     * @param {string} eventName - the name of the event
+     * @param {() => void} callback - the callback function
+     * @returns {void}
+     */
+    emit(eventName: string, callback: () => void): void;
+
+    /**
+     * @description returns details of filter types
+     * @returns {{[name: string]: {[name: string]: string}}} the details of the filter types
+     */
+    getFilterTypes(): {[name: string]: {[name: string]: string}};
+
+    /**
+     * @description set password required for loading or editing a document
+     * @param {string} url - the URL of the document, as sent to the callback
+     * @param [string] password - the password, undefined indicates no password
+     * @returns {void}
+     */
+    setDocumentPassword(url: string, password?: string): void;
+
+    /**
+     * @description get version information of the LOKit process
+     * @returns {{[name:string]:any}} the version info in JSON format
+     */
+    getVersionInfo(): {[name: string]: any};
+
+    /**
+     * @description posts a dialog event for the window with given id
+     * @param {number} windowId - the id of the window to notify
+     * @param {string} args - the arguments for the event
+     * @returns {void}
+     */
+    sendDialogEvent(windowId: number, args: string): void;
+
+    /**
+     * @description loads a given document
+     * @param {string} path - the document path
+     * @returns {DocumentClient} a handle to the document client
+     */
+    loadDocument(path: string): DocumentClient;
+
+    /**
+     * @description run a macro
+     * @param {string} url - the url for the macro (macro:// URI format)
+     * @returns {boolean} success
+     */
+    runMacro(url: string): boolean;
+  }
+}
```

Comment:
_Outside of the namespace_ we also want to add:
```
declare interface Window {
  libreoffice: LibreOffice.OfficeClient;
}
```

This way we can just throw `electron/libreoffice` into a tsconfig's `types` property and it will just work